lib/pure/math.nim

```diff
@@ -1182,6 +1182,14 @@ func lcm*[T](x, y: T): T =
 
   x div gcd(x, y) * y
 
+func clamp*[T](val: T, rng: Slice[T]): T {.since: (1, 5), inline.}=
+  ## Like `system.clamp` but takes a slice so you can easily clamp within a range.
+  runnableExamples:
+    doAssert clamp(10, 1 .. 5) == 5
```

Comment:
https://nim-lang.github.io/Nim/contributing.html

> An exception to the above rule is runnableExamples and code-block rst blocks intended to be used as runnableExamples, which for brevity use assert instead of doAssert